exp/addons/api/v1beta1/clusterresourceset_webhook.go

```diff
@@ -96,6 +96,15 @@ func (m *ClusterResourceSet) validate(old *ClusterResourceSet) error {
 		)
 	}
 
+	// Validate that the strategy is either "ApplyOnce" or "Reconcile".
```

Comment:
Isn't this redundant because we already do this via OpenAPI? https://github.com/kubernetes-sigs/cluster-api/blob/929687253b052aaecc5cfc2da5e9dc86999ce325/config/crd/bases/addons.cluster.x-k8s.io_clusterresourcesets.yaml#L439-L440

Reply:
Wouldn't be the first time that someone has an issue which should be technically impossible if our webhooks would be active :)

Reply:
Although in this case it wouldn't be our webhooks of course but using the right CRDs and a normal kube-apiserver :)